lib/ret_web/channels/session_socket.ex

```diff
@@ -12,10 +12,25 @@ defmodule RetWeb.SessionSocket do
     "session:#{socket.assigns.session_id}"
   end
 
-  def connect(%{"session_id" => session_id}, socket) do
+  def connect(%{"session_token" => session_token}, socket) do
+    session_id =
+      case session_token |> Ret.SessionToken.decode_and_verify() do
```

Comment:
I'd push this little routine into a function on `SessionToken` so you can then just do:
```
assign(:session_id, session_token |> SessionToken.session_id_for_token || SecureRandom.uuid()
```

also might be nice to extract `SecureRandom.uuid()` into a function `generate_session_id` for documentation purposes